### ISSUE TITLE
Improve generation of stubs to allow to generate them in parallel

### ISSUE THREAD

**Issue by activeeon-bot** [ow2-proactive/programming#1043]
<a href="https://jira.activeeon.com/browse/PROACTIVE-1168" title="PROACTIVE-1168">Original issue</a> created by <a href="mailto:bastien.sauvan_AT_sophia.inria.fr">Bastien Sauvan</a> on 24, Nov 2011 at 13:16 PM - PROACTIVE-1168

<hr />

Currently, stubs cannot be generated in parallel in the same JVM because the class JavassistByteCodeStubBuilder is not thread safe. If several stubs are generated in parallel the static fields proxyGetter and proxySetter can be modified concurrently and thus makes the generation fails.
To solve this issue, there is at least 2 solutions:
- make the method createStubObjectMethods synchronized.
- the fileds proxyGetter and proxySetter are static fields only to be used in the method checkMethod where their signatures are used to ensure that there is no conflict with the methods to be reified. Thus, instead of storing the CtMethods proxyGetter and proxySetter, only their signature could be stored.

**Comment by activeeon-bot** [comment]
<a href="https://jira.activeeon.com/browse/PROACTIVE-1168?focusedCommentId=18631">Original comment</a> posted by <a href="mailto:youri.bonnaffe_AT_activeeon.com">Youri Bonnaffe</a> on 12, Dec 2014 at 13:07 PM

<hr />

Marking old issues as won't fix, either because we don't support it anymore, or it is needed or it has been there for too long.

Feel free to reopen if needed.
